apps/teams-test-app/src/App.tsx

```diff
@@ -42,6 +43,35 @@ if (
   }
 }
 
+function handleMessageFromMockedHost(msg: MessageEvent): void {
+  if (!msg || !msg.data) {
```

Comment:
>!msg [](http://example.com/codeflow?start=6&length=4)

Is it necessary to perform this check? When would the event object ever be `undefined` or `null`? The prototype for `addEventListener` doesn't indicate that the event should ever be `undefined`.

Reply:
My initial thought is that, If I don't add this check, it is to say I need to make sure, or we have an implicit agreement, that our testing host WILL NOT send empty message. It would not happen right now but we cannot make sure we don't want to do it in the future. I don't want this function to rely on what testing host does/sends so I added this check.

But now I think it would be better if it fails directly if people trying to pass in empty msg instead of filter it out by if-else condition check so that developer might not get aware of himself sending empty msg. Will remove it.